crates/bitwarden/src/secrets_manager/state.rs

```diff
@@ -0,0 +1,52 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::AccessToken,
+    crypto::{EncString, KeyDecryptable, KeyEncryptable},
+    error::{Error, Result},
+};
+use std::{fmt::Debug, path::Path};
+
+const STATE_VERSION: u32 = 1;
+
+#[cfg(feature = "secrets")]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ClientState {
+    pub(crate) version: u32,
+    pub token: String,
+    pub encryption_key: String,
+}
+
+impl ClientState {
+    pub fn new(token: String, encryption_key: String) -> Self {
+        Self {
+            version: STATE_VERSION,
+            token,
+            encryption_key,
+        }
+    }
+}
+
+pub fn get(state_file: &Path, access_token: &AccessToken) -> Result<Option<ClientState>> {
+    let file_content = std::fs::read_to_string(state_file)?;
+
+    let encrypted_state: EncString = file_content.parse()?;
+    let decrypted_state: String = encrypted_state.decrypt_with_key(&access_token.encryption_key)?;
+    let client_state: ClientState = serde_json::from_str(&decrypted_state)?;
+
+    if client_state.version != STATE_VERSION {
+        return Err(Error::InvalidStateManagerFileVersion);
+    }
+
+    Ok(Some(client_state))
+}
```

Comment:
No, it will either have the state or it will fail, I believe this propagated from previous changes. Good call, resolved in [d0c0c10](https://github.com/bitwarden/sdk/pull/388/commits/d0c0c10dadd1a72550a2eccdd861b1a44c8f1859)